File: Code/Actions/DelayAction.cpp
#include "DelayAction.h"

DelayAction::DelayAction(double seconds){
    DelayAction::seconds = seconds;
    flag = true;
    name = "Delay Action (" + std::to_string(seconds) + " seconds)";
}
GoldRushAction* DelayAction::run(Robot* robot){
    if(flag){
        begin = std::chrono::steady_clock::now();
        flag = false;
    }

    double diff = (std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::steady_clock::now() - begin).count())*0.000001;

    return diff > seconds ? nextAction : this;

}

File: RobotControl/Actions/DelayAction.py
from Actions.GoldRushAction import GoldRushAction
from Robot.Robot import Robot
from datetime import datetime


class DelayAction(GoldRushAction):
    endTime: float

    def __init__(self, delayTime: float, description: str = ..., ) -> None:
        """Construct Delay Action

        Args:
            delayTime (float): Seconds to delay
            description (str, optional): description to print. Defaults to ....
        """
        super().__init__(description)
        self.endTime = -1
        self.delayTime = delayTime
    def run(self, robot: Robot) -> GoldRushAction:
        if(self.endTime == -1):
            self.endTime = (datetime.utcnow().timestamp()) + (self.delayTime * 2)

        curr = (datetime.utcnow().timestamp())
        print(curr, self.endTime)
        if( curr >= self.endTime):
            return self.nextAction
        else:
            return self
        
